subgraphs/euler-finance/src/mappings/helpers.ts

```diff
@@ -162,316 +139,607 @@ export function createRepay(event: Repay): BigDecimal {
+  // These fees should be excluded in calculating total & supply side
+  // revenue. Here we roll back total and supply side revenue that have already
+  // been added because logAssetStatus is emitted before Liquidation in
+  // executeLiquidation().
+  const protocol = getOrCreateLendingProtocol();
+  const market = getOrCreateMarket(marketId);
+  const token = getOrCreateToken(event.params.underlying);
+  const repay = bigIntToBDUseDecimals(event.params.repay, DEFAULT_DECIMALS);
+  // Line 156 Liquidation.sol: repay * (1 + 0.02)
+  const desiredRepay = repay.times(BIGDECIMAL_ONE.plus(UNDERLYING_RESERVES_FEE.div(DECIMAL_PRECISION)));
+  const repayExtraUSD = desiredRepay.minus(repay).times(token.lastPriceUSD!);
+  const deltaTotalRevenueUSD = repayExtraUSD.times(RESERVE_FEE_SCALE).div(assetStatus.reserveFee.toBigDecimal());
+  const deltaSupplySideRevenueUSD = deltaTotalRevenueUSD.minus(repayExtraUSD);
+
+  protocol.cumulativeTotalRevenueUSD = protocol.cumulativeTotalRevenueUSD.minus(deltaTotalRevenueUSD);
+  protocol.cumulativeSupplySideRevenueUSD = protocol.cumulativeSupplySideRevenueUSD.minus(deltaSupplySideRevenueUSD);
+  protocol.save();
+
+  const financialsSnapshot = getOrCreateFinancials(event.block.timestamp, event.block.number);
+  financialsSnapshot.cumulativeTotalRevenueUSD =
+    financialsSnapshot.cumulativeTotalRevenueUSD.minus(deltaTotalRevenueUSD);
+  financialsSnapshot.cumulativeSupplySideRevenueUSD =
+    financialsSnapshot.cumulativeSupplySideRevenueUSD.minus(deltaSupplySideRevenueUSD);
+  if (financialsSnapshot.dailyTotalRevenueUSD.ge(deltaTotalRevenueUSD)) {
+    financialsSnapshot.dailyTotalRevenueUSD = financialsSnapshot.dailyTotalRevenueUSD.minus(deltaTotalRevenueUSD);
+    financialsSnapshot.dailySupplySideRevenueUSD =
+      financialsSnapshot.dailySupplySideRevenueUSD.minus(deltaSupplySideRevenueUSD);
+    financialsSnapshot.save();
+  }
+  financialsSnapshot.save();
+
+  market.cumulativeTotalRevenueUSD = market.cumulativeTotalRevenueUSD.minus(deltaTotalRevenueUSD);
+  market.cumulativeSupplySideRevenueUSD = market.cumulativeSupplySideRevenueUSD.minus(deltaSupplySideRevenueUSD);
+  market.save();
+
+  const marketDailySnapshot = getOrCreateMarketDailySnapshot(event.block, marketId);
+  marketDailySnapshot.cumulativeTotalRevenueUSD =
+    marketDailySnapshot.cumulativeTotalRevenueUSD.minus(deltaTotalRevenueUSD);
+  marketDailySnapshot.cumulativeSupplySideRevenueUSD =
+    marketDailySnapshot.cumulativeSupplySideRevenueUSD.minus(deltaSupplySideRevenueUSD);
+  if (marketDailySnapshot.dailyTotalRevenueUSD.ge(deltaTotalRevenueUSD)) {
+    marketDailySnapshot.dailyTotalRevenueUSD = marketDailySnapshot.dailyTotalRevenueUSD.minus(deltaTotalRevenueUSD);
+    marketDailySnapshot.dailySupplySideRevenueUSD =
+      marketDailySnapshot.dailySupplySideRevenueUSD.minus(deltaSupplySideRevenueUSD);
+    marketDailySnapshot.save();
+  } else {
+    log.error(
```

Comment:
this seems like another dev log, but if you think it should be included then it should also be logged for the hourly snapshot.

Reply:
simplified & added.